auth/src/main/java/org/aerogear/mobile/security/impl/SecurityCheckExecutorImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.aerogear.mobile.security.impl;
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.Check;
+import org.aerogear.mobile.security.SecurityCheck;
+import org.aerogear.mobile.security.SecurityCheckExecutor;
+import org.aerogear.mobile.security.SecurityCheckResult;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.ArrayList;
+
+public class SecurityCheckExecutorImpl implements SecurityCheckExecutor {
+    private final ArrayList<SecurityCheck> checks;
```

Comment:
Can we have the same check more than one time? If no, this should be a Set.
Moreover, it is a good practice to never declare a variable as ArrayList: better to declare it as List.
In our case, I think declaring it as Collection would be the best choice: that way we are able to change the real container without touching the rest of the code.